docs/upgrade.adoc

```diff
@@ -5,6 +5,69 @@
 JanusGraph release.
 
+=== Upgrading to JanusGraph 0.3.0
+
+JanusGraph 0.3.0 implements <<schema-constraints, schema constraints>> which made it necessary to also introduce the concept of a schema version. There is a check to prevent client connections that either expect a different schema version or have no concept of a schema version. To perform an upgrade, the configuration option `graph.allow-upgrade=true` must be set on each graph you wish to upgrade. The graph must be opened with a 0.3.0 or greater version of JanusGraph since older versions have no concept of `graph.storage-version` and will not allow for it to be set.
+
+Example excerpt from `janusgraph.properties` file
+[source, text]
+----
+# JanusGraph configuration sample: Cassandra over a socket
+#
+# This file connects to a Cassandra daemon running on localhost via
+# Thrift.  Cassandra must already be started before starting JanusGraph
+# with this file.
+
+# This option should be removed as soon as the upgrade is complete. Otherwise if this file
+# is used in the future to connect to a different graph it could cause and unintended upgrade.
+graph.allow-upgrade=true
+
+gremlin.graph=org.janusgraph.core.JanusGraphFactory
+
+# The primary persistence provider used by JanusGraph.  This is required.
+# It should be set one of JanusGraph's built-in shorthand names for its
+# standard storage backends (shorthands: berkeleyje, cassandrathrift,
+# cassandra, astyanax, embeddedcassandra, cql, hbase, inmemory) or to the
+# full package and classname of a custom/third-party StoreManager
+# implementation.
+#
+# Default:    (no default value)
+# Data Type:  String
+# Mutability: LOCAL
+storage.backend=cassandrathrift
+
+# The hostname or comma-separated list of hostnames of storage backend
+# servers.  This is only applicable to some storage backends, such as
+# cassandra and hbase.
+#
+# Default:    127.0.0.1
+# Data Type:  class java.lang.String[]
+# Mutability: LOCAL
+storage.hostname=127.0.0.1
+
+----
+
+*IMPORTANT* You should back-up your data prior to attempting an upgrade!
+If `graph.allow-upgrade` is set to true on a graph `graph.storage-version` and `graph.janusgraph-version` will automatically be upgraded to match the version level of the server, or local client, that is opening the graph.
+You can verify the upgrade was successful by opening the management API and validating the values of `graph.storage-version` and `graph.janusgraph-version`.
+[source, gremlin]
+----
+graph = JanusGraphFactory.open("janusgraph.properties");
+// If Using ConfiguredGraphFactory set graph by using ConfiguredGraphFactory.open() as shown below
+// graph = ConfiguredGraphFactory.open("graph-name");
+==>standardjanusgraph[cassandrathrift:[127.0.0.1]]
+mgmt = graph.openManagement()
+==>org.janusgraph.graphdb.database.management.ManagementSystem@44834f50
+mgmt.get("graph.storage-version");
+==>2
+mgmt.get("graph.janusgraph-version");
+==>0.3.0-SNAPSHOT
+mgmt.close()
+==>null
+----
```

Comment:
I think you can remove this code block now since updating the Management API is no longer necessary.